pkg/dryrun/dryrun_utils.go

```diff
@@ -0,0 +1,264 @@
+							mapActualObj) {
+							elementMatch = true
+
+							break
+						}
+					}
+
+					if !elementMatch {
+						allMatch = false
+
+						errorMatchAnyArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+
+						break
+					}
+
+					successMatchArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+				}
+
+				if !allMatch {
+					errorMatchArrayPrint(cmd, path)
+
+					isStatusMatch = false
+				} else {
+					successMatchArrayPrint(cmd, path)
+				}
+			default:
+				// Simple comparison of arrays eg []string
+				if reflect.DeepEqual(typedV, actualArray) {
+					successMatchPrint(cmd, path, typedV, actualArray)
+				} else {
+					errorMatchPrint(cmd, path, typedV, actualArray)
+
+					isStatusMatch = false
+				}
+			}
+
+		case map[string]interface{}:
+			if actualMap, ok := actualValue.(map[string]interface{}); ok {
+				// Recursive call for nested maps
+				isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, typedV, actualMap)
+			} else {
+				failedParsePrint(cmd, path)
+
+				isStatusMatch = false
+			}
+		case interface{}:
+			typeOfTypeV := reflect.TypeOf(typedV).String()
+			typeOfActual := reflect.TypeOf(actualValue).String()
+
+			if typeOfTypeV == "map[interface {}]interface {}" &&
+				typeOfActual == "map[interface {}]interface {}" {
+				mapObj, ok := convertMapStringKey(typedV).(map[string]interface{})
+				if !ok {
+					failedParsePrint(cmd, path)
+
+					continue
+				}
+
+				mapActualObj, ok := convertMapStringKey(actualValue).(map[string]interface{})
+				if !ok {
+					failedParsePrint(cmd, path)
+
+					continue
+				}
+
+				isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, mapObj, mapActualObj)
+
+				continue
+			}
+
+			if typeOfTypeV != typeOfActual {
+				cmd.Println(errorColor(fmt.Sprintf("The path '%s' does not match at all", path)))
+
+				continue
+			}
+
+			cmd.Println(errorColor(fmt.Sprintf("Unsupported type %T at path '%s' for key '%s'", typedV, path, key)))
+		default:
+			cmd.Println(errorColor(fmt.Sprintf("Unsupported type %T at path '%s' for key '%s'", typedV, path, key)))
+		}
+	}
+
+	return isStatusMatch
+}
+
+func failedParsePrint(cmd *cobra.Command, path string) {
+	cmd.Println(warningColor("failed to parse, skip " + path))
+}
+
+func notArrayPrint(cmd *cobra.Command, from, path string) {
+	cmd.Println(warningColor(fmt.Sprintf("skip %s, %s is not array,", path, from)))
+}
+
+func errorMatchPrint(cmd *cobra.Command, path string, inputVal, actualVal interface{}) {
+	cmd.Println(errorColor(fmt.Sprintf("%s: %v does not match %v", path, inputVal, actualVal)))
+}
+
+func errorMatchAnyArrayPrint(cmd *cobra.Command, path string) {
+	cmd.Println(errorColor(fmt.Sprintf("%s does not match any elements", path)))
+}
+
+func errorMatchArrayPrint(cmd *cobra.Command, path string) {
+	cmd.Println(errorColor(fmt.Sprintf("%s does not match the elements", path)))
+}
+
+func successMatchArrayPrint(cmd *cobra.Command, path string) {
+	cmd.Println(successColor(fmt.Sprintf("%s is founded", path)))
```

Comment:
```suggestion
	cmd.Println(successColor(fmt.Sprintf("%s matches", path)))
```

I'm worried this message is very repetitive. Maybe it shouldn't be printed at all?